### PR TITLE
Fix fsevents_file_watcher, add tests

### PR DESCRIPTION
**Pull request by dragonsinth**

### CODE REVIEW

File: google/appengine/tools/devappserver2/mtime_file_watcher_test.py [hunk 1]
```diff
@@ -35,8 +35,8 @@ class TestMtimeFileWatcher(unittest.TestCase):
   """Tests for mtime_file_watcher.MtimeFileWatcher."""
 
   def setUp(self):
-    self._directory = tempfile.mkdtemp()  # The watched directory
-    self._junk_directory = tempfile.mkdtemp()  # A scrap directory.
+    self._directory = os.path.realpath(tempfile.mkdtemp())  # The watched directory
+    self._junk_directory = os.path.realpath(tempfile.mkdtemp())  # A scrap directory.
```
Comment (dragonsinth) [inline google/appengine/tools/devappserver2/mtime_file_watcher_test.py]: this existing test wasn't passing on mac for me due to `/var` vs `/private/var` path/expectation differences; this fixes

File: google/appengine/tools/devappserver2/fsevents_file_watcher.py [hunk 2]
```diff
@@ -13,70 +13,100 @@
...
-                   p not in set(['api', 'checkout', 'platform', 'preorder', 'store']) and
-                   not p.startswith('.') and
-                   p != "assets" and
-                   p != "node_modules" and
-                   p != "resources" and
-                   p != "static" and
-                   p != "test"]
-
-# Only watch for changes to .go or .yaml files
-WATCHED_EXTENSIONS = set(['.go', '.yaml'])
```
Comment (dragonsinth) [inline google/appengine/tools/devappserver2/fsevents_file_watcher.py]: ^^ idk wtf all this is; watched extensions might be useful?

File: google/appengine/tools/devappserver2/fsevents_file_watcher.py [hunk 3]
```diff
@@ -13,70 +13,100 @@
...
-        time.sleep(0.1)
+    def changes(self, timeout=0):
+      try:
+        self._change_event.wait(timeout / 1000.0)
         changed = set(self._changes)
-        del self._changes[:]
         return changed
+      finally:
+        self._changes = []
+        self._change_event.clear()
```
Comment (dragonsinth) [inline google/appengine/tools/devappserver2/fsevents_file_watcher.py]: Basically I sort of slowly pulled in a lot of ideas and code from https://github.com/Khan/frankenserver/blob/master/python/google/appengine/tools/devappserver2/fsevents_file_watcher.py

Where did this impl come from?

Comment (joelgwebber) [reply google/appengine/tools/devappserver2/fsevents_file_watcher.py]: It came from https://gist.github.com/zeekay/5eba991c39426ca42cbb, that Seth found. I originally left that in the commit comment, but it got lost somehow.

**Comment by dragonsinth** [comment]
It kind of seems to work. :D Probably would be good to get more people testing it tho.  Will go ahead and make an internal PR for it.

**Pull request merged**